index.rst

```diff
@@ -5,77 +5,331 @@
+In short, the ``Catcher`` will have an "alarm" event with payloads to indicate the "severity" and other relevant information.
+There will also be commands to mute and acknowledge these alarms.
+We can also consider to borrow the alarm escalation model from the ``Watcher``.
 
 Generate reports
 ----------------
 
-The Catcher should also be able to generate reports, with graphs and performing additional lightweight computation.
-The original idea was to rely on parameterized Jupyter notebooks that would later be converted to webpages and rendered online so users could inspect the results, and use as reference for additional followup.
+The ``Catcher`` should also be able to generate reports, with graphs and additional lightweight computation.
+
+One of the initial ideas was to rely on parameterized Jupyter notebooks that would be processed using a `papermill`_-like application, converted to a webpage and rendered online so users could inspect the results, and use as reference for additional followup.
+
+.. _papermill: https://papermill.readthedocs.io/en/latest/
 
 This could still be achieved using a similar solution or relying on a service like noteburst :sqr:`065` :cite:`SQR-065` and Times Square :sqr:`066` :cite:`SQR-066` to execute parameterized notebooks using a web request and publishing the results afterwards.
 
+Another alternative is to implement a template system, where the user writes a template report with placeholders for text and images.
+Then, the report can be generated by executing code to produce the placeholder values and images and exporting the report.
+The report could be a webpage, a pdf (e.g. generated with LaTeX) or any other electronic format.
 
 Service Architecture
 ====================
 
-The Catcher would be a service built on top of RSP functionalities (some existing and others still under development), with ancillary support from LOVE and OCPS.
-It will mostly rely on `Flux`_ scripts and `InfluxDB task`_ to execute operations based on data streams from the EFD bucket.
-The results can be persisted in the EFD, along with summary information to be displayed by LOVE.
+In summary our proposal is:
+
+*  The ``Catcher`` will be an indexed CSC where each instance handle a single rule.
+*  Rules will adopt the "configuration-based" approach.
+   The ``Catcher`` codebase will have a series of "simple" rules that can be chained together to form an activation flag.
+   The rule is only activated if all conditions are met.
+*  When a rule is activated, it executes an analysis process.
+   The analysis process should generate information about alarm level and data to assemble a report.
+*  A report is generated using the output of the analysis process.
+*  An alarm is issued indicating the severity level and information about how to access the report.
+*  Whether a report is generated and/or an alarm is published every time the rule is activated or only at certain severity levels will be a configuration parameter.
+
```

Comment:
I think we'll need logs of sorts as well... which we get for free using the CSC architecture I think... but maybe worth mentioning that we can utilize that feature?
It's useful because sometimes the condition will trigger, then a calculation may run, then the second evaluation of a criteria will fail so no alert will be processed. In this case it's still useful to log that something kicked off.